File: app/src/main/java/com/kota/Bahamut/Pages/BookmarkPage/BookmarkAdapter.java
package com.kota.Bahamut.Pages.BookmarkPage;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.kota.Bahamut.DataModels.Bookmark;
import com.kota.Bahamut.R;

import java.util.List;

public class BookmarkAdapter extends RecyclerView.Adapter<BookmarkViewHolder> {
    private final List<Bookmark> _bookmarks;
    private BookmarkClickListener mClickListener;

    public BookmarkAdapter(List<Bookmark> dataSet) {
        _bookmarks = dataSet;
    }

    @NonNull
    @Override
    public BookmarkViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.board_extend_optional_page_bookmark_item_view, parent, false);
        return new BookmarkViewHolder(v, mClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull BookmarkViewHolder holder, int position) {
        Bookmark bookmark = getItem(position);
        holder.setBookmark(bookmark);
    }

    public Bookmark getItem(int position) {
        return this._bookmarks.get(position);
    }

    @Override
    public int getItemCount() {
        return _bookmarks.size();
    }

    public void setOnItemClickListener(BookmarkClickListener listener) {
        this.mClickListener = listener;
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/bookmarkPage/BookmarkAdapter.kt
package com.kota.Bahamut.pages.bookmarkPage

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.kota.Bahamut.dataModels.Bookmark
import com.kota.Bahamut.R

class BookmarkAdapter(private val _bookmarks: MutableList<Bookmark?>) :
    RecyclerView.Adapter<BookmarkViewHolder?>() {
    private var mClickListener: BookmarkClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BookmarkViewHolder {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.board_extend_optional_page_bookmark_item_view, parent, false)
        return BookmarkViewHolder(v, mClickListener)
    }

    override fun onBindViewHolder(holder: BookmarkViewHolder, position: Int) {
        val bookmark = getItem(position)
        holder.setBookmark(bookmark)
    }

    fun getItem(position: Int): Bookmark? {
        return this._bookmarks[position]
    }

    override fun getItemCount(): Int {
        return _bookmarks.size
    }

    fun setOnItemClickListener(listener: BookmarkClickListener?) {
        this.mClickListener = listener
    }
}
